### ISSUE TITLE
Parsing error: inferred tsconfigRootDir should be a resolved absolute path in typescript-eslint

### ISSUE THREAD

**Issue by jsdev-robin** [typescript-eslint/typescript-eslint#11542]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [x] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I am encountering the following parsing error when using typescript-eslint:

`Parsing error: inferred tsconfigRootDir should be a resolved absolute path, but received: "C:/Users/User/Documents/workStation/devmun-email-server/server". This is a bug in typescript-eslint! Please report it to us at https://github.com/typescript-eslint/typescript-eslint/issues/new/choose.eslint
`

This occurs when ESLint tries to parse TypeScript files. It seems that typescript-eslint is incorrectly inferring tsconfigRootDir on Windows paths, causing the parser to fail.

<img width="1272" height="61" alt="Image" src="https://github.com/user-attachments/assets/78411689-b2a6-4cdd-9e90-54bec15f6e3b" />

### Fail Cases

```TypeScript
Running ESLint on any .ts or .tsx file in a Windows environment with typescript-eslint configured.

Using a project structure where the ESLint parserOptions.project points to a tsconfig.json inside a subfolder (e.g., server/tsconfig.json).

tsconfigRootDir is not explicitly set, causing typescript-eslint to infer it incorrectly from a Windows-style path.

Projects using monorepo setups or nested tsconfig.json files where relative paths are resolved differently on Windows.
```

### Pass Cases

```TypeScript
ESLint runs successfully on .ts or .tsx files when parserOptions.tsconfigRootDir is explicitly set to an absolute path.

Projects using a single tsconfig.json at the root of the repository, allowing typescript-eslint to correctly infer the root directory.

Running ESLint on Unix-like systems (Linux or macOS) where path resolution works as expected.

Monorepos or nested projects where each tsconfig.json has its own correctly set tsconfigRootDir.

Any TypeScript project where ESLint parses files without relying on automatic inference of tsconfigRootDir.
```

### Additional Info

This issue appears to be specific to Windows environments, where typescript-eslint struggles with backslashes and drive letters in inferred paths.

Explicitly setting parserOptions.tsconfigRootDir to an absolute path can serve as a temporary workaround.

The problem is reported upstream in the [typescript-eslint repository](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose.eslint?utm_source=chatgpt.com)
.

It may affect monorepos or projects with multiple tsconfig.json files nested in subdirectories.

No other errors occur if the tsconfigRootDir is correctly resolved; the issue is only with automatic path inference.
